nccid_cleaning/etl.py

```diff
@@ -0,0 +1,242 @@
+import json
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import pandas as pd
+from pydicom.dataset import Dataset
+from tqdm.auto import tqdm
+
+
+def select_image_files(
+    base_path: Path,
+    select_first: bool = True,
+    select_last: bool = False,
+    select_all: bool = False,
+) -> List[Path]:
+    """
+    Selects the first, last, first & last (alphabetically), or all files
+    from each image subdirectory.
+    Note: `select_all=True` will override `select_first` and `select_last`.
+    """
+    files_to_process = []
+    for dirpath, subdirlist, file_list in tqdm(
+        os.walk(base_path), desc="Finding files"
+    ):
+        selected_files = map(
+            lambda i: (select_all)
+            or (select_first and (i == 0))
+            or (select_last and (i == len(file_list) - 1)),
+            range(len(file_list)),
+        )
+        files_to_process.extend(
+            [
+                Path(dirpath) / file
+                for selected, file in zip(selected_files, sorted(file_list))
+                if selected
+            ]
+        )
+
+    return files_to_process
+
+
+def ingest_dicom_json(file: Path) -> Dataset:
+    """
+    Reads in a single of json file with pydicom
+    """
+
+    with open(file, "rb") as f:
+        # Read {"InlineBinary": null} as {"InlineBinary": b""}
```

Comment:
I wonder if an answer to "why" this is needed would be useful here? 
(also the comment might be better placed before the line where this actually happens?)

Reply:
Needed o parse the "null" entries from DICOM as missing. Comment moved to better location